security/encryption.go

```diff
@@ -4,53 +4,74 @@ import (
 	"crypto/aes"
 	"crypto/cipher"
 	"crypto/rand"
-	"encoding/base64"
+	"encoding/json"
 	"errors"
+	"sort"
+	"strconv"
+	"strings"
 
 	errorsMsg "github.com/cloudtrust/common-service/errors"
 )
 
 // EncrypterDecrypter used to encrypt/decrypt data
 type EncrypterDecrypter interface {
-	Encrypt(value []byte, additional []byte) ([]byte, error)
-	Decrypt(value []byte, additional []byte) ([]byte, error)
+	Encrypt(value []byte, additional []byte) ([]byte, string, error)
+	Decrypt(value []byte, kid string, additional []byte) ([]byte, error)
 }
 
-type aesGcmCrypting struct {
-	key     []byte
-	tagSize int
+type aesGcmKey struct {
+	Kid      string `json:"kid"`
+	Key      []byte `json:"value"`
+	tagSize  int    `json:"-"`
+	priority int    `json:"-"`
 }
 
-// NewAesGcmEncrypter creation from slice of bytes
-func NewAesGcmEncrypter(key []byte, tagSize int) (EncrypterDecrypter, error) {
-	cd := aesGcmCrypting{
-		key:     key,
-		tagSize: tagSize,
-	}
-	return &cd, cd.validate()
+type keyMaterial struct {
+	Keys []aesGcmKey
 }
 
-// NewAesGcmEncrypterFromBase64 creation from base64
-func NewAesGcmEncrypterFromBase64(base64Key string, tagSize int) (EncrypterDecrypter, error) {
-	key, err := base64.StdEncoding.DecodeString(base64Key)
+// NewAesGcmEncrypterFromBase64 creation from json structure serialized as string
+func NewAesGcmEncrypterFromBase64(keys string, tagSize int) (EncrypterDecrypter, error) {
+	// parse key array
+	var keyEntries []aesGcmKey
+	err := json.Unmarshal([]byte(keys), &keyEntries)
```

Comment:
Rather than a base64 encoded value, you could have configuration like this one:
```
keys:
  key_202003:
    tagSize: 12
    priority: 4
    value: AAAAAAAAAAAAAAAAAAA==
  key_202109:
    tagSize: 12
    priority: 2
    value: ABCABCABCABCABCABCA==
```